File: C# ad y ad/2_ Orientado a objetos/S01.1_ Presentacion/Solucion_ejercicio_presentacion/Ejercicio_presentacion/Personaje.cs
using System;


namespace Ejercicio_presentacion
{
	class Personaje
	{
		string nombre;
		int numero;


		public Personaje(string nombre, int numero)
		{
			this.nombre = nombre;
			this.numero = numero;
		}


		public void Presentarse()
		{
			Console.WriteLine("¡Hola! ¡Me llamo " + nombre + "! Participo en la solución de este ejercicio.");
		}
		public void DecirNumeroGuardado()
		{
			Console.WriteLine("El número guardado de " + nombre + " es " + numero + ".");
		}
		public void Despedirse()
		{
			Console.WriteLine(nombre + " se despide. ¡Hasta luego!");
		}
	}
}

File: C++ ad y ad/3_ Orientado a objetos/S01.1_ Presentacion/Solucion_ejercicio_presentacion/Ejercicio_presentacion/personaje.cpp
#include "personaje.h"


Personaje::Personaje(string nombre, int numero)
{
	this->nombre = nombre;
	this->numero = numero;
}
Personaje::~Personaje()
{
	cout << "El personaje " << nombre << " ha sido liberado de la memoria.\n\n";
}


void Personaje::presentarse()
{
	cout << "Hola! Me llamo " << nombre << "! Participo en la solucion de este ejercicio.\n";
}
void Personaje::decirNumeroGuardado()
{
	cout << "El numero guardado de " << nombre << " es " << numero << ".\n";
}
void Personaje::despedirse()
{
	cout << nombre << " se despide. Hasta luego!\n";
}
